src/operator/controllers/intents/reconciler.go

```diff
@@ -0,0 +1,336 @@
+	pod := obj.(*v1.Pod)
+	otterizeIdentity, err := r.serviceIdResolver.ResolvePodToServiceIdentity(context.Background(), pod)
+	if err != nil {
+		logrus.Errorf("Failed resovling Otterize identity for pod %s: %v", pod.Name, err)
+	}
+	fullClientName := fmt.Sprintf("%s.%s", otterizeIdentity.Name, otterizeIdentity.Namespace)
+	logrus.Infof("Enqueueing client intents for client '%s'", fullClientName)
+
+	clientIntentsList := otterizev1alpha3.ClientIntentsList{}
+	err = r.client.List(ctx,
+		&clientIntentsList,
+		&client.MatchingFields{OtterizeIntentsClientIndexNameField: fullClientName},
+	)
+
+	if err != nil {
+		logrus.Errorf("Failed to list intents for client %s: %v", fullClientName, err)
+	}
+	requests := make([]reconcile.Request, 0)
+	for _, clientIntents := range clientIntentsList.Items {
+		request := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      clientIntents.Name,
+				Namespace: clientIntents.Namespace,
+			},
+		}
+		requests = append(requests, request)
+	}
+
+	return requests
+}
+
+func (r *Reconciler) mapPGServerConfToClientIntents(ctx context.Context, obj client.Object) []reconcile.Request {
+	pgServerConf := obj.(*otterizev1alpha3.PostgreSQLServerConfig)
+	logrus.Infof("Enqueueing client intents for PostgreSQLServerConfig change %s", pgServerConf.Name)
+
+	intentsToReconcile := make([]otterizev1alpha3.ClientIntents, 0)
+	intentsList := otterizev1alpha3.ClientIntentsList{}
+	dbInstanceName := pgServerConf.Name
+	err := r.client.List(ctx,
+		&intentsList,
+		&client.MatchingFields{otterizev1alpha3.OtterizeTargetServerIndexField: dbInstanceName},
+	)
+	if err != nil {
+		logrus.Errorf("Failed to list client intents targeting %s: %v", dbInstanceName, err)
+	}
+	intentsToReconcile = append(intentsToReconcile, intentsList.Items...)
+
+	requests := make([]reconcile.Request, 0)
+	for _, clientIntents := range intentsToReconcile {
+		requests = append(requests, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      clientIntents.Name,
+				Namespace: clientIntents.Namespace,
+			},
+		})
+	}
+	return requests
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var intents otterizev1alpha3.ClientIntents
+	err := r.client.Get(ctx, req.NamespacedName, &intents)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, errors.Wrap(err)
+	}
+
+	dbNames := extractDBInstanceNames(intents)
+	if len(dbNames) == 0 {
+		return ctrl.Result{}, nil
+	}
+
+	for _, databaseName := range dbNames {
+		pgServerConfigs := otterizev1alpha3.PostgreSQLServerConfigList{}
+		err := r.client.List(ctx, &pgServerConfigs)
+		if err != nil {
+			r.recorder.Eventf(&intents, v1.EventTypeWarning, ReasonErrorFetchingPostgresServerConfig,
+				"Error trying to fetch '%s' PostgresServerConf for client '%s'. Error: %s",
+				databaseName, intents.GetServiceName(), err.Error())
+			return ctrl.Result{}, errors.Wrap(err)
+		}
```

Comment:
List the pgServerConfigs outside the for loop (see similar comment on the intents-operator PR)